extensions/default/src/DicomTagBrowser/DicomTagTable.tsx

```diff
@@ -1,97 +1,203 @@
+        <label
+          ref={vrRef}
+          className="flex flex-col flex-1 text-white text-lg pl-1 select-none"
+        >
+          <span className="flex flex-row items-center focus:outline-none">
+            VR
+          </span>
+        </label>
+      </div>
+      <div className="px-3 w-6/24">
+        <label
+          ref={keywordRef}
+          className="flex flex-col flex-1 text-white text-lg pl-1 select-none"
+        >
+          <span className="flex flex-row items-center focus:outline-none">
+            Keyword
+          </span>
+        </label>
+      </div>
+      <div className="px-3 w-5/24 grow">
+        <label
+          ref={valueRef}
+          className="flex flex-col flex-1 text-white text-lg pl-1 select-none"
+        >
+          <span className="flex flex-row items-center focus:outline-none">
+            Value
+          </span>
+        </label>
       </div>
     </div>
   );
 }
 
 function DicomTagTable({ rows }) {
+  const listRef = useRef();
+  const canvasRef = useRef();
+
+  /**
+   * When new rows are set, scroll to the top and reset the virtualization.
+   */
+  useEffect(() => {
+    if (!listRef?.current) {
+      return;
+    }
+
+    listRef.current.scrollTo(0);
+    listRef.current.resetAfterIndex(0);
+  }, [rows]);
+
+  /**
+   * When the browser window resizes, update the row virtualization (i.e. row heights)
+   */
+  React.useEffect(() => {
+    function handleResize() {
+      listRef.current.resetAfterIndex(0);
+    }
+
+    window.addEventListener('resize', handleResize);
+
+    return () => {
+      window.removeEventListener('resize', handleResize);
+    };
+  });
+
+  const Row = ({ index, style }) => {
+    const row = rows[index];
+
+    return (
+      <div
+        style={{ ...style, ...rowStyle }}
+        className={classNames(
+          'hover:bg-secondary-main transition duration-300 bg-black flex flex-row w-full border-secondary-light items-center text-base break-all',
+          lineHeightClassName
+        )}
+        key={`DICOMTagRow-${index}`}
+      >
+        <div className="px-3 w-4/24">{row[0]}</div>
+        <div className="px-3 w-2/24">{row[1]}</div>
+        <div className="px-3 w-6/24">{row[2]}</div>
+        <div className="px-3 w-5/24 grow">{row[3]}</div>
+      </div>
+    );
+  };
+
+  const [tagHeaderElem, setTagHeaderElem] = useState(null);
+  const [vrHeaderElem, setVrHeaderElem] = useState(null);
+  const [keywordHeaderElem, setKeywordHeaderElem] = useState(null);
+  const [valueHeaderElem, setValueHeaderElem] = useState(null);
+
+  const tagRef = elem => {
+    if (elem) {
+      setTagHeaderElem(elem);
+    }
+  };
+  const vrRef = elem => {
+    if (elem) {
+      setVrHeaderElem(elem);
+    }
+  };
+  const keywordRef = elem => {
+    if (elem) {
+      setKeywordHeaderElem(elem);
+    }
+  };
+  const valueRef = elem => {
+    if (elem) {
+      setValueHeaderElem(elem);
+    }
+  };
+
+  const isHeaderRendered = () => tagHeaderElem != null;
```

Comment:
no need to recreate this on every render, just use tagHeaderElemen !== null in the return

Reply:
I will change to a `useCallback`